src/include/interval_set.h

```diff
@@ -764,6 +764,22 @@ class interval_set {
     return std::move(m);
   }
 
+  /*
+  * Round down interval starts and round up interval ends alignment the specified @alignment
+  */
+  void align(T alignment) {
```

Comment:
(not sure about this, but:) shouldn't 'alignment' be of the diff type of 'T', not 'T' itself?

Reply:
Within interval_set, T must behave like n integer. As such I think operator- must be overloaded as T operator-(T, T). 

This assumption is made in many places, so I think it's safe to keep the assumption here.